src/ListGroup.js

```diff
@@ -13,17 +13,15 @@ class ListGroup extends React.Component {
 
     if (!this.props.children) {
       return this.renderDiv(items);
-    } else if (React.Children.count(this.props.children) === 1 && !Array.isArray(this.props.children)) {
-      let child = this.props.children;
-
-      childrenAnchors = this.isAnchor(child.props);
-
     } else {
+      React.Children.forEach(this.props.children, (child) => {
+        if (!Array.isArray(child) && this.isAnchor(child.props)) {
+          childrenAnchors = true;
+        }
 
-      childrenAnchors = Array.prototype.some.call(this.props.children, (child) => {
-        return !Array.isArray(child) ? this.isAnchor(child.props) : Array.prototype.some.call(child, (subChild) => {
-            return this.isAnchor(subChild.props);
-        });
+        if (Array.isArray(child) && child.some.call((subChild) => this.isAnchor(subChild.props))) {
```

Comment:
It's for this test case: https://github.com/react-bootstrap/react-bootstrap/blob/v0.23.7/test/ListGroupSpec.js#L77-L89

But `React.Children.forEach` unrolls that for you so you don't need to manually nest the iterations any more.